Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ContentGraph.php

```diff
@@ -188,6 +189,19 @@ public function findParentNodeAggregates(
         return $this->mapQueryBuilderToNodeAggregates($queryBuilder);
     }
 
+    public function findAncestorNodeAggregateIds(NodeAggregateId $entryNodeAggregateId): NodeAggregateIds
```

Comment:
As far a I understood, this function needs to be more complex and return a set of pathways with ancestor nodes.
https://neos-project.slack.com/archives/C04PYL8H3/p1725605274115779?thread_ts=1725460465.658809&cid=C04PYL8H3

Reply:
Hmm but to get that right, does that mean that if implemented more complex also other node ids will be returned or just that all the node ids will be nicely ordered by dimension in a special value object. The latter i consider just an improvement and not critical. The idea is to improve the performance of the method via native sql at first. Its marked as `@internal` so we can improve the structure later?

Reply:
Yes this one:
> the node ids will be nicely ordered by dimension in a special value object

> Hmm but to get that right, does that mean that if implemented more complex also other node ids will be returned or just that all the node ids will be nicely ordered by dimension in a special value object. The latter i consider just an improvement and not critical. The idea is to improve the performance of the method via native sql at first. Its marked as @internal so we can improve the structure later?

But you don't improve that with this PR. 
And even if it is marked as internal, i think we shouldn't put that into an interface  if we already know, that we will change that.

Reply:
> But you don't improve that with this PR.

i _could_ try to do that ... emphasis on try :D or ask bernhard. But i get your point. Lets see how we can continue.

cc @nezaniel @bwaidelich @kitsunet

Reply:
added behat tests and introduced cte query to improve performance. Its now ordered desc by the parentnodeanchor